### PR TITLE
Checkouts Controller Unit Testing

### PR DESCRIPTION
**Pull request by LevinGeorge**

### CODE REVIEW

File: spec/controllers/checkouts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    before :each do
+      user = create(:user)
+      sign_in user
+      @shop = create(:shop, :user => user)
+    end
+
+    it "goes to the checkout page" do
+      product = create(:product, shop: @shop)
+      get :new, id: product.id
+      cards = create(:card, :user => subject.current_user)
```
Comment (IonatanSala) [inline spec/controllers/checkouts_controller_spec.rb]: @LevinGeorge Why are you creating a card here? You don't need to create a card to go to the checkout page.

File: spec/controllers/checkouts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe "POST#checkout_current_card" do
```
Comment (IonatanSala) [inline spec/controllers/checkouts_controller_spec.rb:25]: @LevinGeorge You should also check that after the charge or customer is made on the stripe server, that we also save   the relevant information in our database on the customer or the charge.

**Comment by danielpaul** [comment]
@LevinGeorge @IonatanSala could you guys have a look at this and see what still need to be done and get this across the line? Same as the other PR. Also need to resolve merge conflicts on this.

File: spec/controllers/checkouts_controller_spec.rb (hunk 1 above)
Comment (LevinGeorge) [inline spec/controllers/checkouts_controller_spec.rb]: :+1:

File: spec/controllers/checkouts_controller_spec.rb (hunk 2 above)
Comment (LevinGeorge) [inline spec/controllers/checkouts_controller_spec.rb:25]: @IonatanSala Can the customer information and charge be accessed from `stripe_customer_id` and `stripe_card_id`?

Comment (LevinGeorge) [inline spec/controllers/checkouts_controller_spec.rb:25]: `expect { Stripe::Charge.create(customer: 'Brian',amount: 1, currency: 'eur') }.to eq(stripe_customer_id && stripe_card_id)` something like this? @IonatanSala

Comment (LevinGeorge) [inline spec/controllers/checkouts_controller_spec.rb:25]: Or maybe something like this? `expect { Stripe::Charge.create(customer: 'Brian',amount: 1, currency: 'eur') }.to be_present` :P

Comment (LevinGeorge) [inline spec/controllers/checkouts_controller_spec.rb:25]: 3rd option..  
`expect(charge.customer).to be_present
 expect(charge.amount).to be_present
 expect(charge.currency).to be_present`

Idk if this is accurate but it seems to be working @IonatanSala  :P

**Comment by LevinGeorge** [comment]
@danielpaul Ready for this to merge now..

**Pull request merged**